From cf683912b378a906d398fead361315b5a34e5c9c Mon Sep 17 00:00:00 2001
From: LN Liberda <lauren@selfisekai.rocks>
Date: Tue, 17 Feb 2026 14:20:44 +0100
Subject: [PATCH] Revert "Update fsanitizer=array-bounds config"

Unknown argument as of clang 21.1.8.

This reverts commit acb47d9a6b56c4889a2ed4216e9968cfc740086c.
---
 build/config/compiler/BUILD.gn                       | 12 ------------
 .../blink/renderer/platform/wtf/text/string_impl.h   |  8 ++++----
 2 files changed, 4 insertions(+), 16 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index be740d17b8346..5b38bae851dcf 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -1912,12 +1912,6 @@
 ubsan_hardening("c_array_bounds") {
   sanitizer = "array-bounds"
   condition = !(is_asan && target_cpu == "x86")
-
-  # Because we've enabled array-bounds sanitizing we also want to suppress
-  # the related warning about "unsafe-buffer-usage-in-static-sized-array",
-  # since we know that the array bounds sanitizing will catch any out-of-
-  # bounds accesses.
-  cflags = [ "-Wno-unsafe-buffer-usage-in-static-sized-array" ]
 }
 
 # Enable UBSan subset to ensure we always hit `return` statement in non-void
